src/charm.py

```diff
@@ -130,24 +132,37 @@ def reconcile(self, event: EventBase) -> None:
         if getattr(event, "departing_unit", None) == self.unit:
             return
 
+        outdated_status = []
         # Maintain the correct app status
         if self.unit.is_leader():
             if self.state.opensearch_server:
-                clear_status(self.app, MSG_DB_MISSING)
+                outdated_status.append(MSG_DB_MISSING)
 
         # Maintain the correct unit status
-        if self.state.cluster.tls and not self.state.unit_server.tls:
-            self.unit.status = MaintenanceStatus(MSG_TLS_CONFIG)
-        else:
-            clear_status(self.unit, MSG_TLS_CONFIG)
 
+        # Request new certificates if IP changed
+        if self.state.cluster.tls:
+            if self.state.unit_server.tls and self.tls_manager.certificate_valid():
+                outdated_status.append(MSG_TLS_CONFIG)
+            else:
+                self.unit.status = MaintenanceStatus(MSG_TLS_CONFIG)
+
+        # Restart on config change
         if (
             self.config_manager.config_changed()
             and self.state.unit_server.started
             # and self.upgrade_events.idle
         ):
+            # NOTE: We never clean up MSG_RECONFIG, as it MUST be replaced by rolling_ops statuses
+            # (Otherwise the server is out-of-date)
+            # It is to ensure continuality of non-active status as needed
+            self.unit.status = MaintenanceStatus(MSG_RECONFIG)
```

Comment:
I don't insist on this (as VERY soon the charm goes into `rolling_ops` restart phases anyawy.)

If this is raising a debate I rather prefer to get rid of if.

Reply:
We can get rid of it.

Reply:
Done.